### DISCUSSION TITLE
Duplicate: Value in react is not being used as the source of truth

### DISCUSSION THREAD

**Discussion by stevemarksd** [ianstormtaylor/slate#4824] [General]
https://github.com/ianstormtaylor/slate/blob/80661509ecf39b5d8256fa387c7eff15f60bf612/packages/slate-react/src/components/slate.tsx#L17

I've been playing with this for a while, but seems like you need to edit the state directly on the `Editor` instead of just changing the `value` from the use state.

```
  const [value, setValue] = useState<Descendant[]>([])

<Editor
        value={this.state.value}
...
```

if you call `setValue` with your own data, is just ignored, the data doesn't update in the editor, this is a bit weird. Could we make the source of truth in react or is not possible?

What do you think?

**Comment by dylans** [comment]
See https://github.com/ianstormtaylor/slate/pull/4540 and https://github.com/ianstormtaylor/slate/pull/4768

**Reply by stevemarksd** [reply]
Cool, thanks and sorry for not checking that up in the first place.
